src/other/broadcasting.jl

```diff
@@ -0,0 +1,63 @@
+struct LazyNewColDataFrame
+    df::DataFrame
+    col::Symbol
+end
+
+# we allow LazyNewColDataFrame only for data frames with some columns
+Base.axes(x::LazyNewColDataFrame) = axes(x.df[1])
+
+Base.Broadcast.broadcastable(df::AbstractDataFrame) = Matrix(df)
+
+Base.maybeview(df::AbstractDataFrame, idxs...) = view(df, idxs...)
+
+function Base.maybeview(df::AbstractDataFrame, idxs)
+    if ncol(df) == 0
+        throw(ArgumentError("Broadcasting into a data frame with no columns is not allowed"))
+    end
+    if idxs isa Symbol
+        if !haskey(df, idxs)
+            if !(df isa DataFrame)
+                throw(ArgumentError("column $idxs not found"))
+            end
+            return LazyNewColDataFrame(df, idxs)
+        end
+    end
+    view(df, idxs)
+end
+
+function Base.copyto!(lazydf::LazyNewColDataFrame, bc)
+    if isempty(lazydf.df)
+        throw(ArgumentError("Broadcasting creation of a column of an empty data frame is not allowed"))
+    end
+    T = mapreduce(i -> typeof(bc[i]), promote_type, eachindex(bc); init=Union{})
+    col = Tables.allocatecolumn(T, nrow(lazydf.df))
+    copyto!(col, bc)
+    lazydf.df[lazydf.col] = col
+end
+
+function Base.copyto!(df::AbstractDataFrame, bc)
+    for I in eachindex(bc)
+        # a safeguard against linear index
+        row, col = Tuple(I)
+        df[row, col] = bc[I]
+    end
+    df
+end
+
+function Base.copyto!(df::AbstractDataFrame, bc::Base.Broadcast.Broadcasted{<:Base.Broadcast.AbstractArrayStyle{0}})
+    if bc.f === identity && bc.args isa Tuple{Any} && Base.Broadcast.isflat(bc)
```

Comment:
This is equivalent, right?
```suggestion
    if bc.f === identity && bc.args isa Tuple && Base.Broadcast.isflat(bc)
```

Can you add a comment to explain what this condition means? Same for the other branch.

Reply:
Your suggestion is not equivalent I think. In general this is a verbatim copy from Base giving a condition when we can perform a faster operation as we can get a scalar we want to broadcast over directly.

Reply:
Ah, right, `Tuple{Any}` implies length 1.

Reply:
I will add a comment in the code though.